frontend/src/queries/Query/Query.tsx

```diff
@@ -134,7 +134,9 @@ export function Query<Q extends Node>(props: QueryProps<Q>): JSX.Element | null
                         <>
                             <QueryEditor
                                 query={JSON.stringify(query)}
-                                setQuery={(stringQuery) => setQuery?.(JSON.parse(stringQuery))}
+                                setQuery={(stringQuery) =>
+                                    setQuery?.(JSON.parse(stringQuery), /* isSourceUpdate= */ true)
```

Comment:
```suggestion
                                    setQuery?.(JSON.parse(stringQuery), true)
```
IDE hints should be enough